BlueprintUI/Sources/Layout/Builder/StackElement+Builder.swift

```diff
@@ -0,0 +1,115 @@
+/// and then pull of the given traits and key and then apply those to the stack
+public struct StackChild: ProxyElement {
+    public var traits: StackLayout.Traits
+    public var key: AnyHashable?
+    private let wrappedElement: Element
+
+    public init(
+        wrappedElement: Element,
+        traits: StackLayout.Traits = .init(),
+        key: AnyHashable? = nil
+    ) {
+        self.wrappedElement = wrappedElement
+        self.traits = traits
+        self.key = key
+    }
+    
+    public init(
+        wrappedElement: Element,
+        growPriority: CGFloat = 1.0,
+        shrinkPriority: CGFloat = 1.0,
+        alignmentGuide: ((ElementDimensions) -> CGFloat)? = nil,
+        key: AnyHashable? = nil
+    ) {
+        self.init(
+            wrappedElement: wrappedElement,
+            traits: .init(
+                growPriority: growPriority,
+                shrinkPriority: shrinkPriority,
+                alignmentGuide: alignmentGuide.map(StackLayout.AlignmentGuide.init)
+            ),
+            key: key
+        )
+    }
+
+    // Simply wraps the given element.
+    public var elementRepresentation: Element { wrappedElement }
+}
+
+extension Element {
+    
+    /// Wraps an element with a `StackChild` in order to customize `StackLayout.Traits` and the key.
+    /// - Parameters:
+    ///   - growPriority: Controls the amount of extra space distributed to this child during underflow.
+    ///   - shrinkPriority: Controls the amount of space allowed for this child during overflow.
+    ///   - alignmentGuide: Allows for custom alignment of a child along the cross axis.
+    ///   - key: A key used to disambiguate children between subsequent updates of the view
+    ///     hierarchy.
+    /// - Returns: A wrapped element with additional layout information for the `StackElement`.
+    public func stackChild(
+        growPriority: CGFloat = 1.0,
+        shrinkPriority: CGFloat = 1.0,
+        alignmentGuide: ((ElementDimensions) -> CGFloat)? = nil,
+        key: AnyHashable? = nil
+    ) -> StackChild {
+        .init(
+            wrappedElement: self,
+            growPriority: growPriority,
+            shrinkPriority: shrinkPriority,
+            alignmentGuide: alignmentGuide, key: key
+        )
+    }
+    
+    /// Wraps an element with a `StackChild` in order to customize `StackLayout.Traits` and the key.
+    /// - Parameters:
+    ///   - traits: Contains traits that affect the layout of individual children in the stack.
+    ///   - key: A key used to disambiguate children between subsequent updates of the view
+    ///     hierarchy.
+    /// - Returns: A wrapped element with additional layout information for the `StackElement`.
+    public func stackChild(traits: StackLayout.Traits = .init(), key: AnyHashable? = nil) -> StackChild {
+        .init(wrappedElement: self, traits: traits, key: key)
+    }
+}
+
+
+extension StackElement {
```

Comment:
Thought: The two options presented for configuring the stack's options have so far been:

1) The `configure` block option added here
2) The option you described in the PR description, where we add "builder-like" callbacks that let you do `.horizontalUnderflow(.center)` on the stack to configure these options.

A third option that in retrospect, seems really obvious, is just including these params in the `init` as optional before the builder closure, eg:

```
public init(
   horizontalUnderflow : ... = .default,
   horizontalOverflow : ... = .default,
   @Builder<Element> elements: () -> [Element],
)
```

Reply:
I like this idea, but I wonder about scalability. We would likely need to put every instance variable into the initializer. Is that ok? As the property list grows the init could become unwieldy, though a lot of that will be hidden via default values. 

We could also consider a hybrid approach...add the most commonly used properties to the init and then use the configure block or builder-functions to customize the rest.

As a separate note: one thing that detracts me from the configure block is the possibility of mixing the `@Builder` approach with `add(Element:)`. This could lead to an unclear API (which gets called first? configure or the builder?).

Reply:
Coming back to this after paternity leave and I think leaning on the initializer makes the most sense...default values help keep it clean and it avoids the boiler plate/maintenance of adding all of the modifiers. I will update the PR so we can see how it looks in practice.

Reply:
Welcome back!

Reply:
Ok, I updated the syntax to remove configure blocks and instead add configurations to the initializer. An example looks like this:

````swift
Row(horizontalUnderflow: .center) {
    Element1()
    Element2()
        .stackChild(growPriority: 0.0)
    Element3()
}
````